core/src/confidence.rs

```diff
@@ -55,26 +62,22 @@ impl ForkConfidenceCache {
 
     pub fn get_fork_with_depth_confidence(
         &self,
-        minimum_depth: usize,
-        minimum_stake_percentage: f64,
+        confidence_config: ConfidenceConfig,
     ) -> Option<u64> {
         self.bank_confidence
             .iter()
             .filter(|&(_, bank_confidence)| {
-                let fork_stake_minimum_depth: u64 = bank_confidence.confidence[minimum_depth..]
+                let fork_stake_minimum_depth: u64 = bank_confidence.confidence
+                    [confidence_config.confirmations - 1..]
                     .iter()
                     .cloned()
                     .sum();
                 fork_stake_minimum_depth as f64 / self.total_stake as f64
-                    >= minimum_stake_percentage
+                    >= confidence_config.percentage
```

Comment:
Instread of [the failed solution of using ulp](https://github.com/solana-labs/solana/pull/6538/files#r339320086), how about writing like this?:

```rust
pub const DEFAULT_PERCENT_CONFIDENCE: f64 = 2_f64 / 3_f64;
...
                fork_stake_minimum_depth as f64 / self.total_stake as f64
                    > confidence_config.percentage // note the `>`, not `>=`

```

(I'm assuming this code needs some formality because we're defining the exact definition of _confidence_ here)

Reply:
Those seem like fair changes. Thanks!